Index: build/build.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>windows-1252
===================================================================
--- build/build.properties	(revision 2151)
+++ build/build.properties	(revision )
@@ -1,4 +1,4 @@
-package.name = DBN-9.0.zip
-idea.version = 9.0
-idea.home= C:/Program Files/JetBrains/IntelliJ IDEA ${idea.version}
-idea.plugins.sandbox=T:/IntelliJIdea/${idea.version}/system/plugins-sandbox${idea.version}/plugins
\ No newline at end of file
+package.name = DBN-10.0.zip
+idea.version = 10.0
+idea.home= C:/Program Files (x86)/JetBrains/IntelliJ IDEA 10
+idea.plugins.sandbox=D:/IntelliJIdea/${idea.version}/system/plugins-sandbox${idea.version}/plugins
\ No newline at end of file
Index: src/com/dci/intellij/dbn/vfs/DatabaseFileViewProvider.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>windows-1252
===================================================================
--- src/com/dci/intellij/dbn/vfs/DatabaseFileViewProvider.java	(revision 2151)
+++ src/com/dci/intellij/dbn/vfs/DatabaseFileViewProvider.java	(revision )
@@ -8,6 +8,7 @@
 import com.intellij.psi.PsiFile;
 import com.intellij.psi.PsiManager;
 import com.intellij.psi.SingleRootFileViewProvider;
+import com.intellij.testFramework.LightVirtualFile;
 import org.jetbrains.annotations.NotNull;
 
 public class DatabaseFileViewProvider extends SingleRootFileViewProvider {
@@ -49,7 +50,7 @@
 
     @NotNull
     @Override
-    public SingleRootFileViewProvider createCopy(VirtualFile copy) {
+    public SingleRootFileViewProvider createCopy(LightVirtualFile copy) {
         return new DatabaseFileViewProvider(getManager(), copy, false, getBaseLanguage());
     }
 
